### DISCUSSION TITLE
Add a RSO integration

### DISCUSSION THREAD

**Discussion by DarkIntaqt** [TheDrone7/shieldbow#49] [Ideas]
Add a RSO integration for shieldbow. I do not expect shieldbow to launch a custom webserver which is collecting the callback, but maybe something like this
```ts
// I coded that in the GitHub preview editor, please ignore any issues within the code, this should just be an example
const rso_config = {
    client_id: String,
    client_secret: String,
    callback_url: String (probably also URL object),
    response_type: String ("code" [default]),
    SCOPE: String ("openid" [default], "cpid", "offline_access")
};

client.rso.generateSignInUrl(rso_config); // https://auth.riotgames.com/authorize?client_id=CLIENTID&redirect_uri=CALLBACK_URL&response_type=RESPONSE_TYPE&scope=SCOPE

// and then something like this, I dunno, just an idea
client.rso.getUserData(token[s], rso_config); // prints puuid or so

client.rso.refreshToken(token[s], rso_config); // im not sure if rso is even capable of this, that's just an example
```

**Comment by TheDrone7** [comment]
I feel like this is heavily unnecessary, in most cases, you'd probably be using something like passport.js or next auth and since RSO is ultimately pretty much just oauth they should be able to handle it much better and should allow for other login methods in case people might want that in their projects.

**Reply by DarkIntaqt** [reply]
You might be right that this sounds a bit "overpowered". Still, I think adding at least the possibility to use /riot/account/v1/accounts/me would be pretty helpful to get a users puuid
